server/src/main/resources/rules/rules.js

```diff
@@ -338,7 +337,7 @@ function get_pool_priority(pool, consumer) {
     }
 
     // Decrease the priority of shared pools slightly so that non-shared pools will get consumed first.
-    if (Utils.equalsIgnoreCase('true', pool.getAttribute(SHARE_DERIVED))) {
+    if (Utils.equalsIgnoreCase('true', pool.hasSharedAncestor.toString())) {
```

Comment:
If this is a boolean value, why are we converting it to a string to check if it's true?